### PR TITLE
Disable parallel paginated search for multi base dn on subsequent pages

### PR DESCRIPTION
**Pull request by butonic**
When multiple base dns are configured the `AllUsersIterator` uses pages of 500 to page through the ldap results. It literally provides limit and offset:

https://github.com/owncloud/core/blob/d2697818a682904a13ba8d18e96411722853dd8a/lib/private/User/Sync/AllUsersIterator.php#L40-L47

That confuses the internal parralel seach cookie handling:
1. the first base dn tree only contains ... say 3 results.
2. the second base dn tree contains 1000
the first one will have no further results after the first page, causing the search to be abandoned -> which means all cookies are removed. even for the second base dn tree.

while that shouldn't matter, the next page starts with offset 500.

again, the first base dn tree has no results, abandoning the search and deleting all cookies ... so the sync has no chance of iterating over all pages.

something like that ... It is not 100% correct because it is hard to follow the code path. I need to build a test to cover this automatically.

### CODE REVIEW

**Comment by PVince81** [comment]
no reviewers, assuming still WIP ?

**Comment by PVince81** [comment]
@butonic please specify what needs to be done to finish this so we can take over

**Comment by micbar** [comment]
@butonic Could you rebase and clarify, what needs to be done?

**Comment by micbar** [comment]
Please rebase and merge.

Seems that this is working. Tested by customers.

**Comment by micbar** [comment]
Is Access.php Unit testable?

**Comment by VicDeo** [comment]
@micbar 
> Is Access.php Unit testable?

I'm afraid it's too complex.

**Comment by mrow4a** [comment]
@micbar @VicDeo closing -> Detected while writing integration tests that this fix is conceptually wrong. Lets assume we request users with limit 2 and offset 2. With this fix, and e.g. 3 base DNs, there is possibility that we will return 3x2users, 2 for each base DN, thus exceeding limit of 2!

**Comment by mrow4a** [comment]
we discussed with @butonic and this is expected behaviour -> add comments in the code and retest.

**Comment by mrow4a** [comment]
@micbar @jvillafanez ready for rereview -> I am pretty confident on paging in LDAP with this PR merged. 

Tested with integration tests, but also manually:
- in UI counting with multiple base dns in various configurations
- syncing users, also with changing user attributes in ldap server and sync
- searching for users

**Review by jvillafanez — APPROVED** [review]
We might need some additional comments in the search functions to warn that it's possible to get results over the limit. Other than that, the code looks fine.

**Comment by mrow4a** [comment]
@jvillafanez will add the comments asap

**Review by jvillafanez — APPROVED** [review]
